documentation/meta/traffic/runbooks/identifying-and-blocking-traffic-anomalies.md

```diff
@@ -0,0 +1,161 @@
+
+## When to use this runbook
+
+Use this runbook when either of these are true:
+
+1. There is a service disruption to the Openverse API or frontend.
+2. There is a marked increase in resource usage on the API or frontend ECS
+   services that continues to trend to unsustainable levels, which _could lead_
+   to a service disruption.
+
+And:
+
+1. There is a marked increase in traffic visible in our AWS dashboards or
+   Cloudflare analytics.
+
+## Requirements
+
+1. Access to the Openverse Cloudflare dashboard.
+2. Familiarity with the Cloudflare web UI, including the Analytics, Security,
+   and Web Access Firewall tabs.
+3. Time. This is an exploratory process that can take significant trial-and
+   error.
+
+## Procedure Steps
+
+### 1. Identify the timeframe of the traffic anomalies
+
+1.  Log into Cloudflare.
+2.  Select the relevant domain in the Cloudflare UI.
+3.  Go to `Analytics & Logs => Traffic` and choose the "Page Views" box in the
+    left sidebar.
+4.  Set the time range dropdown to 24 hours.
+5.  See if you can find the point where the current spike in traffic begins.
+6.  If you do not see a place where traffic stabilizes to a relatively flat
+    line, continue increasing the time range in the dropdown.
+7.  Once you find where normalized traffic starts to spike in the graph, hover
+    over the graph and **record the date and time the spike began**.
+
+### 2. Identify targetable sources of the suspicious traffic
+
+Next, we need to associate the traffic with specific sources. Essentially, we're
+trying to find unique identifiers linked to potentially malicious traffic, so we
+can block them through the Cloudflare Web Access firewall.
+
+There are three primary identifiers for targeting traffic. These are ordered
+from largest, most persistent, and most impactful, to smallest and most
+ephemeral. When possible, **blocking ASNs** is the single best way to mitigate
+malicious traffic.
+
+#### Traffic Source Identifiers
+
+- Source ASNs (Autonomous System Numbers): Unique identifiers for a group of IP
+  networks and routers under the control of a single organization, typically an
+  internet service provider or hosting company.
+
+  Typically, a single ASN makes up less than 10% of total Openverse requests in
+  a given timeframe. If this percentage is larger, it could suggest that a user
+  on the ASN is scraping Openverse or accessing Openverse programmatically in an
+  aggressive way.
+
+- Source user agents: Unique user-agent strings sent with requests to Openverse.
+  Nearly all UA strings from web browsers begin with `Mozilla`. Outside of the
+  latest stable version of Chrome on Windows 10, which is the most popular
+  Openverse browser (and usually looks something like
+  `Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/{webkit-version} (KHTML, like Gecko) Chrome/{real-chrome-version} Safari/{webkit-version}`),
+  no single user agent should take up more than 10% of total traffic. Excluding
```

Comment:
> no single user agent should take up more than 10% of total traffic

Won't this become less true over time as UA masking becomes more common? Where does this 10% number come from? Same question for the ASN 10% above as well. When would we know to adjust the expectation of "10%" specifically? Is it possible for us to create a dashboard that shows a rolling distribution of these so that we know what to benchmark against at any given moment?

Reply:
Creating a dashboard could be a good idea. For now I've added a note recommending users look at a time slice of traffic _without_ any known malicious traffic or incidents to compare the percentages.

Reply:
Rewording it might be fine for now but it is still a bandaid until we have ongoing observability of what the expected rate should be. It probably won't be an issue for a little while still, UA masking isn't that aggressive yet, which is why it isn't a blocker to me.